addons/mass_mailing/wizard/mailing_contact_import.py

```diff
@@ -0,0 +1,121 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import api, fields, models, tools, _
+from odoo.tools.misc import clean_context
+
+
+class MailingContactImport(models.TransientModel):
+    _name = 'mailing.contact.import'
+    _description = 'Mailing Contact Import'
+
+    @api.model
+    def default_get(self, fields_list):
+        vals = super(MailingContactImport, self).default_get(fields_list)
+        active_id = self._context.get('active_id')
+        if active_id and self.env.context.get('active_model') == 'mailing.mailing':
+            active_mailing = self.env['mailing.mailing'].browse(active_id)
+            if active_mailing.contact_list_ids:
+                vals['list_ids'] = active_mailing.contact_list_ids[0]
+        return vals
+
+    list_ids = fields.Many2many('mailing.list', string='Lists')
+    contact_list = fields.Text('Contact List', help='Contact list that will be imported, one contact per line')
+
+    def action_import(self):
+        """Import each lines of "contact_list" as a new contact."""
+        self.ensure_one()
+
+        contacts = tools.email_split_tuples(self.contact_list)
+        if not contacts:
+            return {
+                'type': 'ir.actions.client',
+                'tag': 'display_notification',
+                'params': {
+                    'message': _('No contacts were imported. All email addresses were invalid.'),
+                    'type': 'warning',
+                    'sticky': False,
+                    'next': {'type': 'ir.actions.act_window_close'},
+                }
+            }
+
+        existing_contacts = self.env['mailing.contact'].search([
+            ('email_normalized', 'in', [values[1].lower() for values in contacts]),
+            ('list_ids', 'in', self.list_ids.ids),
+        ])
+        existing_contacts = {
+            contact.email_normalized: contact
+            for contact in existing_contacts
+        }
+
+        # Remove duplicated record, keep only the first non-empty name for each email address
+        unique_contacts = {}
+        for name, email in contacts:
+            email = email.lower()
+            if unique_contacts.get(email, {}).get('name'):
+                continue
+            if email in existing_contacts and self.list_ids != existing_contacts[email].list_ids:
```

Comment:
Can't we use ``>`` (or ! ``<`` ) with record sets, to tell "if new list ids is not included in existing list ids" ?